src/prefect/core/flow.py

```diff
@@ -568,30 +568,54 @@ def chain(self, *tasks: Task, validate: bool = None) -> List[Edge]:
             )
         return edges
 
-    def update(self, flow: "Flow", validate: bool = None) -> None:
+    def update(
+        self, flow: "Flow", parmerge: bool = None, validate: bool = None
+    ) -> None:
         """
-        Take all tasks and edges in another flow and add it to this flow
-
+        Take all tasks and edges in another flow and add it to this flow.  
+        When `parmerge` is set to`True` -- Duplicate parameters in the unput `flow`
+        are replaced with those in the flow being updated. 
+    
         Args:
-            - flow (Flow): A flow which is used to update this flow
-            - validate (bool, optional): Whether or not to check the validity of the flow
-
+            - flow (Flow): A flow which is used to update this flow.
+            - parmerge (bool, optional): Merge two graphs on parameters
+            - validate (bool, optional): Whether or not to check the validity of the flow.
+        
         Returns:
             - None
         """
+
         for task in flow.tasks:
-            if task not in self.tasks:
+
+            duped_task = self.get_tasks(slug=task.slug) if parmerge else list()
+
+            # Adds a task if slugs are not duplicate
+            if (task not in self.tasks) and (not duped_task):
                 self.add_task(task)
 
+        # Append the new edges to this flow.
         for edge in flow.edges:
-            if edge not in self.edges:
-                self.add_edge(
-                    upstream_task=edge.upstream_task,
-                    downstream_task=edge.downstream_task,
-                    key=edge.key,
-                    mapped=edge.mapped,
-                    validate=validate,
-                )
+
+            if any(
```

Comment:
I need to think more about this edge manipulation logic - we'll need to move away from slugs (for reasons unrelated to this PR)

Reply:
@cicdw Perhaps this PR needs to wait until you finish with `slugs`. #2531? Or is the issue with `slugs` larger than that?

Reply:
@gryBox yup just that PR 👍

Reply:
Well #2531 broke that edge logic :) .  I need to investigate further and understand how to use the new slugs.  Its not obvious to me why Parameters have a slug and Tasks do ... but only if they are set by the user.  Except to ensure parameter uniqueness.